resources/views/website/layouts/macros/upcomingevents.blade.php

```diff
@@ -28,27 +28,31 @@
     </div>
     <div class="card-body">
 
-        @php($events = Proto\Models\Event::where('secret', false)->where('is_featured', false)->where('end', '>=', date('U'))->orderBy('start')->limit($n)->get())
+        @php($events = Proto\Models\Event::where('is_featured', false)->where('end', '>=', date('U'))->orderBy('start')->limit($n)->get())
 
         @if(count($events) > 0)
 
             @foreach($events as $key => $event)
 
-                @include('event.display_includes.event_block', ['event'=> $event])
+                @if(!$event->secret || (Auth::check() && ($event->activity && $event->activity->isParticipating(Auth::user()))))
```

Comment:
Maybe also add a condition for the board permission so they can always see the hidden activities like in the agenda.

Reply:
I have purposefully not done this as the board may want to show their homepage without spoiling any secret events they are not going to. They can instead still see them all at the overview